### PR TITLE
kubernetes deployment Option Change and add function to call maintain class.

### PR DESCRIPTION
**Pull request by ydye**

### CODE REVIEW

File: kubernetes-deployment/bootstrap.py [hunk 1]
```diff
@@ -395,47 +414,54 @@ def remove_nodes(cluster_config, node_list_config):
...
+    # Todo: load maintain from a DB such as etcd instead of a yaml file.
+    maintain_config = load_yaml_file("maintain.yaml")
+
+    for host in node_list_config['machinelist']:
+
+        maintain_one_node(cluster_config, maintain_config, node_list_config['machinelist'][host], job_name)
+
+
+
+def option_validation(args):
```
Comment (fanyangCS) [inline kubernetes-deployment/bootstrap.py]: >option_validation [](start = 4, length = 17)

please add ut. please construct typical invalid input and see if it return false and the expected error hint

File: kubernetes-deployment/maintainlib/common.py [hunk 2]
```diff
@@ -79,12 +79,12 @@ def write_generated_file(generated_file, file_path):
...
-    hostip = host_config['hostip']
-    username = host_config['username']
-    password = host_config['password']
+    hostip = str(host_config['hostip'])
+    username = str(host_config['username'])
+    password = str(host_config['password'])
     port = 22
     if 'sshport' in host_config:
-        port = host_config['sshport']
+        port = int(host_config['sshport'])
```
Comment (fanyangCS) [inline kubernetes-deployment/maintainlib/common.py]: >port [](start = 8, length = 4)

please check whether port value is valid.

File: kubernetes-deployment/maintainlib/common.py [hunk 3]
```diff
@@ -79,12 +79,12 @@ def write_generated_file(generated_file, file_path):
 
 def sftp_paramiko(src, dst, filename, host_config):
 
-    hostip = host_config['hostip']
-    username = host_config['username']
-    password = host_config['password']
+    hostip = str(host_config['hostip'])
```
Comment (fanyangCS) [inline kubernetes-deployment/maintainlib/common.py]: >hostip [](start = 4, length = 6)

please check the format of hostip is valid

File: kubernetes-deployment/maintainlib/common.py [hunk 4]
```diff
@@ -79,12 +79,12 @@ def write_generated_file(generated_file, file_path):
 
 def sftp_paramiko(src, dst, filename, host_config):
```
Comment (fanyangCS) [inline kubernetes-deployment/maintainlib/common.py]: >host_config [](start = 38, length = 11)

I suggest you add a function to validate the format of host_config, and use the funcation here and in ssh_shell_paramiko

[the remark this answers is not in the document]
Comment (ydye) [reply kubernetes-deployment/maintainlib/common.py]: Another feature, do it in the feature in our plan.

File: kubernetes-deployment/maintainlib/common.py (hunk 3 above)
Comment (ydye) [reply kubernetes-deployment/maintainlib/common.py]: Another feature, do it in the feature in our plan.

File: kubernetes-deployment/maintainlib/common.py (hunk 2 above)
Comment (ydye) [reply kubernetes-deployment/maintainlib/common.py]: Another feature, do it in the feature in our plan.

Comment (ydye) [reply kubernetes-deployment/maintainlib/common.py]: By the way, I will remove most function in bootstrap. And instead, I will call them from maintainlib.common.

File: kubernetes-deployment/bootstrap.py (hunk 1 above)
Comment (ydye) [reply kubernetes-deployment/bootstrap.py]: Done

File: kubernetes-deployment/maintainlib/common.py (hunk 2 above)
Comment (fanyangCS) [reply kubernetes-deployment/maintainlib/common.py]: so you prefer to do ut after the code refactoring? 

---
In reply to: [170410130](https://github.com/Microsoft/pai/pull/330#discussion_r170410130) [](ancestors = 170410130)

Comment (ydye) [reply kubernetes-deployment/maintainlib/common.py]: Yes

**Pull request merged**